app/services/importer/get_workday_fields.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+module Importer
+  class GetWorkdayFields < ApplicationTask
+    # Some Workday companies require a login cookie to fetch fields
+    # If login is required, the API response will include an authentication widget
+    # #user_authenticated? checks this and logs in using ENV credentials if required
+    # WorkdayCookieFetcher retrieves the cookie, which is attached to the request header
+    # Then we make the request again, attaching the cookie
+    include FaradayHelpers
+
+    def initialize(api_url)
+      @api_url = api_url
+      @data = fetch_data_from_api
+      @api_url_base = extract_base_url
+      @cookie = nil
+    end
+
+    def call
+      return unless processable
+
+      process
+    end
+
+    private
+
+    def processable
+      @data && user_authenticated?
+    end
+
+    def process
+      build_fields
+      print_and_return_fields # just for testing
+    end
+
+    def authenticate
+      Rails.cache.fetch(cache_key, expires_in: 1.hour) do
```

Comment:
@patanj101 are you also happy with this solution? Other thought was to get multiple keys and rotate through

Reply:
I think caching and multiple usernames are both good ideas: caching to save time, multiple usernames to avoid getting flagged. Can implement both. Is the .env file the best place to store the usernames/passwords?